src/figuresClasses.ts

```diff
@@ -1,19 +1,90 @@
+type Color = 'red'|'green'|'blue';
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
 export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea: Function;
+}
+
+function roundedArea(area: number): number {
+  return Math.floor(area * 100) / 100;
+}
+
+function isTriangle(a:number, b:number, c:number):boolean {
+  const allSides = [a, b, c].sort((x:number, y:number) => x - y);
 
+  return allSides[2] < allSides[0] + allSides[1];
 }
 
 export class Triangle {
+  public shape = Shape.Triangle;
 
+  constructor(
+    public color:Color,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || !isTriangle(a, b, c)) {
+      throw new Error('triangle with such sides does not exist');
+    }
```

Comment:
It would be better to divide Errors by 2 types:
1. All sides are positive (first `if` statement)
2. Triangle is valid (second `if` statement with `!isTriangle`)